src/main/java/org/radarcns/redcap/listener/TokenManagerListener.java

```diff
@@ -44,54 +41,38 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(TokenManagerListener.class);
 
     private static final String ACCESS_TOKEN = "TOKEN";
-    private static final String AUTHORIZATION_HEADER = "Authorization";
-    private static final String SCOPE_HEADER = "scope";
-
-    private static final OkHttpClient client;
-    private static final Request request;
 
-    private static long tokenDuration;
-    private static long lastRefresh;
+    private static final OAuth2Client client;
+    private static OAuth2AccessToken token;
 
     static {
-        client = new OkHttpClient.Builder()
-            .connectTimeout(10, TimeUnit.SECONDS)
-            .writeTimeout(10, TimeUnit.SECONDS)
-            .readTimeout(30, TimeUnit.SECONDS)
-            .authenticator((route, response) -> response.request().newBuilder()
-                    .header(AUTHORIZATION_HEADER, Properties.getOauthCredential())
-                    .build()
-            ).build();
-
         try {
-            request = new Request.Builder()
-                .url(Properties.getTokenEndPoint())
-                .post(new FormBody.Builder().add("grant_type", "client_credentials")
-                    //TODO check if scope is correct. This token must be able to access projects
-                    // and create users
-                    .add(SCOPE_HEADER, "read")
-                    .build())
-                .build();
-        } catch (MalformedURLException exc) {
-            LOGGER.error("Properties cannot be load. Check the log for more information.", exc);
+            client = new OAuth2Client()
+                        .managementPortalUrl(Properties.getTokenEndPoint().toString())
```

Comment:
Is not better to use `URL` here instead of `String`?

Reply:
Indeed, `URL` is safer here to use. I updated the client to have a `tokenEndpoint` instead of `managementPortalUrl`, as this is more general.